ansible_base/rbac/api/serializers.py

```diff
@@ -112,17 +117,18 @@ def get_by_ansible_id(self, ansible_id, requesting_user, for_field):
             raise ValidationError({for_field: msg.format(pk_value=ansible_id)})
         return resource.content_object
 
-    def get_actor_from_data(self, validated_data, requesting_user):
+    def validate(self, attrs):
+        """Validate that exactly one of actor or actor_ansible_id is provided"""
         actor_aid_field = f'{self.actor_field}_ansible_id'
-        if validated_data.get(self.actor_field) and validated_data.get(actor_aid_field):
+
+        # Check what was actually provided in the request
+        has_actor_in_request = self.actor_field in self.initial_data
+        has_actor_aid_in_request = actor_aid_field in self.initial_data
+
+        if (has_actor_in_request and has_actor_aid_in_request) or (not has_actor_in_request and not has_actor_aid_in_request):
```

Comment:
Can this be simplified to:
```suggestion
        # If both actor and actor_ansible_id are present or both not present than we error out
        if has_actor_in_request == has_actor_aid_in_request:
```

Reply:
so smart